doc/tutorials/gapi/face_beautification/face_beautification.markdown

```diff
@@ -0,0 +1,388 @@
+<a href="https://github.com/opencv/open_model_zoo/tree/master/models/intel
+/face-detection-adas-0001">face-detection-adas-0001</a>
+and the
+<a href="https://github.com/opencv/open_model_zoo/blob/master/models/intel
+/facial-landmarks-35-adas-0002/description/facial-landmarks-35-adas-0002.md">
+facial-landmarks-35-adas-0002</a>.
+
+The face detector takes the input image and returns a blob with the shape
+[1, 1, N, 7] after the inference, where `N = 200` is the maximum number of
+faces which can be detected. Obviously, the output `cv::Mat` post-processing is
+required. The structure of the output for every face is the following:
+`[image_id, label, conf, x_min, y_min, x_max, y_max]`,
+all the seven elements are floating point. For our further goals we need to take
+only results with `image_id > 0` because a negative value in this field
+indicates the end of detections; also we can cut detections by the `conf` field
+to avoid mistakes of the detector. The last four float numbers we've got denote
+normalized coordinates and are between 0 and 1; to get the real pixel
+coordinates we should multiply it by the image sizes respectively to the
+directions.
+
+Then, to create masks we need to get landmarks of facial elements for every
+detected face. The chosen detector expects to obtain square area with only one
+face contained so we have to infer the model separately for every ROI we've
+got from the previous detection. The output of the facial landmarks detector is
+a blob with 35 landmarks: the first 18 of them are facial elements
+(eyes, eyebrows, a nose, a mouth) and the last 17 --- a jaw contour. Landmarks
+are floating point values of coordinates normalized relatively to an input ROI
+(not the original frame). In addition, for the further goals we need contours of
+eyes, mouths, faces, etc., not the landmarks. So, post-processing of the Mat is
+also required here. The process is split into two parts --- landmarks'
+coordinates denormalization to the real pixel coordinates of the source frame
+and getting necessary closed contours based on these coordinates.
+
+The last step of processing the inference data is drawing masks using the
+received contours. Honestly speaking, there is no need to get the most accurate
+contours in the world: we can always blur masks borders by gaussian filter to
+occupy a bit more area. Another point that should be mentioned here is getting
+three masks (for areas to be smoothed, for ones to be sharpened and for the
+background) which have no intersections with each other; this approach allows to
+apply the calculated masks to the corresponding images prepared beforehand and
+then just to summarize them to get the output image without any other actions.
+
+All the operations described above are supported fascinatingly by the G-API
+functionality, so this algorithm is appropriate to illustrate G-API usage
+convenience and efficiency in the context of solving a real CV/DL problem.
```

Comment:
This can be removed

Reply:
Yes, mb there are too much words; however, all the text above was about the algorithm in general without any metions of G-API, and I feel the necessity to make a bridge to G-API usage rather than just to start describing G-API features..